src/ToDoContext.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-param-reassign */
+// eslint-disable-next-line
+import _ from 'lodash';
+import React, { useEffect, useReducer } from 'react';
+import { ToDo } from './types/ToDo';
+
+export enum ACTIONS {
+  ADD = 'addPost',
+  REMOVE = 'removePost',
+  UPDATE = 'updatePost',
+  TOGGLE = 'completed',
+  SORT = 'sortBy',
+  TOGGLE_ALL = 'TOGGLE_ALL',
+  CLEAR = 'removeComplited',
+
+}
+
+type Action = { type: ACTIONS.ADD, payload: string }
+| { type: ACTIONS.UPDATE, payload: ToDo }
+| { type: ACTIONS.REMOVE, payload: ToDo }
+| { type: ACTIONS.TOGGLE, payload: number }
+| { type: ACTIONS.SORT, payload: string }
+| { type: ACTIONS.TOGGLE_ALL, payload: boolean }
+| { type: ACTIONS.CLEAR };
+
+interface State {
+  list: ToDo[];
+  sortBy: string;
+}
+
+function newToDo(name: string): ToDo {
+  return { id: Date.now(), title: name, completed: false };
+}
+
+function toggle(list: ToDo[], id: number): ToDo[] {
+  const copyList = _.cloneDeep(list);
+
+  copyList.forEach(todo => {
+    if (todo.id === id) {
+      todo.completed = !todo.completed;
+    }
+  });
+
+  return copyList;
+}
+
+function toggleAllHelper(elem: ToDo, trigger: boolean) {
+  if (!trigger) {
+    elem.completed = true;
+  } else {
+    elem.completed = false;
+  }
+
+  return elem;
+}
+
+function update(list: ToDo[], toDo: ToDo): ToDo[] {
+  const copyList = _.cloneDeep(list);
+  const index = copyList.findIndex(todo => todo.id === toDo.id);
+
+  copyList.splice(index, 1, toDo);
+
+  return copyList;
+}
+
+function remove(list: ToDo[], toDo: ToDo): ToDo[] {
+  const copyList = _.cloneDeep(list);
+  const index = copyList.findIndex(todo => todo.id === toDo.id);
+
+  copyList.splice(index, 1);
+
+  return copyList;
+}
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'addPost':
+      return {
+        ...state,
+        list: [...state.list, newToDo(action.payload)],
+      };
+    case 'updatePost':
+      return {
+        ...state,
+        list: update(state.list, action.payload),
+      };
+    case 'removePost':
+      return {
+        ...state,
+        list: remove(state.list, action.payload),
+      };
+    case 'completed':
```

Comment:
you created the enum but don't use it
replace strings in `switch` by Enum values